### PR TITLE
core: remove manual renaming and let script handle rook to koor renaming

### PR DESCRIPTION
**Pull request by ideepika**
- rook-to-koor.sh: script for resolving patching rook to koor changes, this uses setting up a worktree with latest rook/master and mangling it to adopt koor conventions, creating a commit, before final merging.

Signed-off-by: Deepika Upadhyay <deepika@koor.tech>

**Description of your changes:**

**Which issue is resolved by this Pull Request:**
Resolves #

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://docs.koor.tech/docs/latest/Contributing/development-flow/#commit-structure)).
- [ ] **Skip Tests for Docs**: If this is only a documentation change, add the label `skip-ci` on the PR.
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://docs.koor.tech/docs/latest/Contributing/development-flow/#submitting-a-pull-request)
- [ ] [Pending release notes](https://github.com/koor-tech/koor/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

File: rook-to-koor.sh [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+# places where complete replacement are apt
+sed -i "s|Rook|Koor Storage Distribution|g" ${ROOKDIR}/.github/PULL_REQUEST_TEMPLATE.md \
+${ROOKDIR}/CODE_OF_CONDUCT.md \
+${ROOKDIR}/Documentation/Contributing/development-flow.md \
+${ROOKDIR}/README.md
+
+# release version mangling 1.10 and 1.9 -> 1.0
+sed -i "s/release\-[0-9].[0-9]/release\-1\.0/g" ${ROOKDIR}/Documentation/Contributing/development-flow.md
+sed -i 's/v1\.10\../1\.0\.0/g' ${ROOKDIR}/Documentation/Upgrade/rook-upgrade.md
```
Comment (galexrt) [inline rook-to-koor.sh:50]: Maybe in the future we can use some `git` commands to get the latest tag of Koor, not needed for this PR.

**Comment by gauravsitlani** [comment]
hey @ideepika the Codespell and ShellCheck tests are failing here, can you please look into it ?

**Comment by gauravsitlani** [comment]
```
Error: ./rook-to-koor.sh:60: conflit ==> conflict
```

There's a spelling mistake here.

**Review by gauravsitlani — CHANGES_REQUESTED** [review]
```
Error: ./rook-to-koor.sh:60: conflit ==> conflict
```

There's a spelling mistake here.

File: rook-to-koor.sh [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+#setup a temporary rook latest folder
+git fetch rook master
+git worktree add "${ROOKDIR}" --checkout rook/master
+cd "${ROOKDIR}"
+git checkout -b wip-temp-rook-merge
+
+# ordering matters here
+find "${ROOKDIR}" -type f ! -path "./.git/*" -exec sed -i -e 's|rook/rook.github.io|koor-tech/docs.koor.tech|g' \
+-e 's|rook/rook/issues|koor-tech/koor/issues|g' \
```
Comment (galexrt) [inline rook-to-koor.sh:20]:
```suggestion
find "${ROOKDIR}" -type f ! -path "./.git/*" -exec sed -i \
    -e 's|rook/rook.github.io|koor-tech/docs.koor.tech|g' \
    -e 's|rook/rook/issues|koor-tech/koor/issues|g' \
```

**Review by galexrt — APPROVED** [review]

**Comment by galexrt** [comment]
@gauravsitlani PTAL

File: rook-to-koor.sh [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    -e 's|/work/rook|/work/koor|g' \
+    -e 's|rook/rook|koor/koor-tech|g' \
+    -e 's|[Rook.io Slack](https://slack.rook.io/)|[GitHub Discussions](https://github.com/koor-tech/koor/discussions)|g' \
+    -e 's|Rook version|Koor Storage Distribution version|g' \
+    -e 's|rook.io/docs/rook|docs.koor.tech/docs|g' \
+    -e 's|"$GITHUB_REPOSITORY_OWNER" = "rook"|"$GITHUB_REPOSITORY_OWNER" = "koor-tech"|g' \
+    -e 's|rook/ceph|koorinc/ceph|g' \
+    -e 's|rook.chart|koor.chart|g' \
+    -e 's|version of Rook|version of Koor Storage Distribution|g' \
+    -e 's/Rook*$ website/Koor website/g' \
```
Comment (zalsader) [inline rook-to-koor.sh:31]: What does this mean?

**Review by zalsader — APPROVED** [review]
Minor changes / questions

File: rook-to-koor.sh [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+sed -i "s/release\-[0-9].[0-9]/release\-1\.0/g" "${ROOKDIR}"/Documentation/Contributing/development-flow.md
+sed -i 's/v1\.10\../1\.0\.0/g' "${ROOKDIR}"/Documentation/Upgrade/rook-upgrade.md
+sed -i 's/v1\.9\../v1\.\0\.0/g' "${ROOKDIR}"/Documentation/Upgrade/rook-upgrade.md
+
+# others
+sed -i "s|rook|koor-tech|g" "${ROOKDIR}"/.docs/macros/includes/main.py
+sed -i "s|koorinc/ceph|Koor Storage Distribution|g" "${ROOKDIR}"/Documentation/Troubleshooting/common-issues.md
+
+# apply patch and merge
+# cd ${ROOKDIR} && git apply --reject --whitespace=fix ../rook-to-koor.patch
```
[the remark this answers is not in the document]
Comment (ideepika) [reply rook-to-koor.sh]: removed, thanks!

File: rook-to-koor.sh (hunk 3 above)
Comment (ideepika) [reply rook-to-koor.sh]: updated, as it didn't work, thanks!

Comment (ideepika) [reply rook-to-koor.sh]:
```-  members. The build process will publish the documentation to the [Rook website](https://rook.io/docs/rook/latest/).
-* All storage providers are added to the [Rook.io website](https://rook.io/)
+  members. The build process will publish the documentation to the [Koor website](https://rook.io/docs/rook/latest/).
+* All storage providers are added to the [Koor website](https://rook.io/)```

**Review by gauravsitlani — APPROVED** [review]

**Pull request merged**